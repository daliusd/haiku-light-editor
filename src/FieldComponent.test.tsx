import React from 'react';

import { Editor, ImageField, Settings, Editable } from '.';
import { render, screen, fireEvent } from '@testing-library/react';

describe('FieldComponent', () => {
  const settings: Settings = {
    width: 100,
    height: 100
  };

  const editable: Editable = {
    width: 80,
    height: 80
  };

  const imageField: ImageField = {
    type: 'image',
    id: '1',
    x: 10,
    y: 10,
    width: 60,
    height: 60,
    angle: 0,
    scale: 1,
    cx: 0,
    cy: 0,
    imageWidth: 60,
    imageHeight: 60,
    imageUrl: '/test.png',
    imageFlip: false,
    imageRotation: 0
  };

  it('image can be moved using mouse', () => {
    render(
      <Editor
        settings={settings}
        editable={{ ...editable, fields: [imageField] }}
      />
    );
    expect(screen.getByTestId('editor')).toMatchSnapshot();

    const imageFieldElement = screen.getByTestId('field-1');
    fireEvent.mouseDown(imageFieldElement, { clientX: 30, clientY: 30 });
    fireEvent.mouseMove(imageFieldElement, { clientX: 40, clientY: 40 });
    fireEvent.mouseUp(imageFieldElement, { clientX: 40, clientY: 40 });

    expect(screen.getByTestId('editor')).toMatchSnapshot();
  });

  it('image can be moved using touch', () => {
    render(
      <Editor
        settings={settings}
        editable={{ ...editable, fields: [imageField] }}
      />
    );
    expect(screen.getByTestId('editor')).toMatchSnapshot();

    const imageFieldElement = screen.getByTestId('field-1');
    fireEvent.touchStart(imageFieldElement, {
      changedTouches: [{ clientX: 30, clientY: 30 }]
    });
    fireEvent.touchMove(imageFieldElement, {
      changedTouches: [{ clientX: 41, clientY: 42 }],
      touches: []
    });
    fireEvent.touchEnd(imageFieldElement, {
      changedTouches: [{ clientX: 41, clientY: 42 }]
    });

    expect(screen.getByTestId('editor')).toMatchSnapshot();
  });
});
