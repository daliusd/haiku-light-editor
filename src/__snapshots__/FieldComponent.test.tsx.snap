// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldComponent image can be moved using mouse 1`] = `
<div
  class="editor"
  data-testid="editor"
  style="width: 100px; height: 100px; background: rgb(238, 238, 238);"
>
  <div
    style="width: 80px; height: 80px; left: 10px; top: 10px; position: absolute; background: rgb(255, 255, 255);"
  >
    <div
      data-testid="field-1"
      style="position: absolute; top: 10px; left: 10px; width: 60px; height: 60px; transform: rotate(0rad);"
    >
      <img
        alt=""
        src="/test.png"
        style="width: 60px; height: 60px; position: absolute; left: 0px; top: 0px; transform: scaleX(1) rotate(0deg) translate(0px, 0px);"
      />
    </div>
  </div>
</div>
`;

exports[`FieldComponent image can be moved using mouse 2`] = `
<div
  class="editor"
  data-testid="editor"
  style="width: 100px; height: 100px; background: rgb(238, 238, 238);"
>
  <div
    style="width: 80px; height: 80px; left: 10px; top: 10px; position: absolute; background: rgb(255, 255, 255);"
  >
    <div
      data-testid="field-1"
      style="position: absolute; top: 20px; left: 20px; width: 60px; height: 60px; transform: rotate(0rad);"
    >
      <img
        alt=""
        src="/test.png"
        style="width: 60px; height: 60px; position: absolute; left: 0px; top: 0px; transform: scaleX(1) rotate(0deg) translate(0px, 0px);"
      />
    </div>
  </div>
</div>
`;

exports[`FieldComponent image can be moved using touch 1`] = `
<div
  class="editor"
  data-testid="editor"
  style="width: 100px; height: 100px; background: rgb(238, 238, 238);"
>
  <div
    style="width: 80px; height: 80px; left: 10px; top: 10px; position: absolute; background: rgb(255, 255, 255);"
  >
    <div
      data-testid="field-1"
      style="position: absolute; top: 10px; left: 10px; width: 60px; height: 60px; transform: rotate(0rad);"
    >
      <img
        alt=""
        src="/test.png"
        style="width: 60px; height: 60px; position: absolute; left: 0px; top: 0px; transform: scaleX(1) rotate(0deg) translate(0px, 0px);"
      />
    </div>
  </div>
</div>
`;

exports[`FieldComponent image can be moved using touch 2`] = `
<div
  class="editor"
  data-testid="editor"
  style="width: 100px; height: 100px; background: rgb(238, 238, 238);"
>
  <div
    style="width: 80px; height: 80px; left: 10px; top: 10px; position: absolute; background: rgb(255, 255, 255);"
  >
    <div
      data-testid="field-1"
      style="position: absolute; top: 22px; left: 21px; width: 60px; height: 60px; transform: rotate(0rad);"
    >
      <img
        alt=""
        src="/test.png"
        style="width: 60px; height: 60px; position: absolute; left: 0px; top: 0px; transform: scaleX(1) rotate(0deg) translate(0px, 0px);"
      />
    </div>
  </div>
</div>
`;
